flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/BigQueryDeserializationSchema.java

```diff
@@ -54,10 +59,19 @@ public interface BigQueryDeserializationSchema<IN, OUT>
      * @param record The IN document to de-serialize.
      * @param out The collector to put the resulting messages.
      */
-    default void deserialize(IN record, Collector<OUT> out) throws IOException {
+    default void deserialize(IN record, Collector<OUT> out) throws BigQueryConnectorException {
         OUT deserialize = deserialize(record);
-        if (deserialize != null) {
-            out.collect(deserialize);
+        try {
+            if (deserialize != null) {
+                out.collect(deserialize);
+            }
+        } catch (RuntimeException e) {
```

Comment:
Again, why RuntimeException? Why not cover all `Exception`s?

Reply:
Changed to `Exception`